cluster/src/main/scala/io/snappydata/gemxd/SparkSQLExecuteImpl.scala

```diff
@@ -352,7 +386,15 @@ object SparkSQLExecuteImpl {
           val sb = new StringBuilder()
           Utils.dataTypeStringBuilder(other, sb)(col)
           // write the full length as an integer
-          hdos.writeUTF(sb.toString(), true, false)
+          val utf8String = UTF8String.fromString(sb.toString())
```

Comment:
Can we avoid adding this unneeded conversion back and forth? This can have significant perf implications for queries returning decent number of rows. Did you try using some other typeId (other than VARCHAR/CHAR/CLOB)?